viadot/tasks/customer_gauge.py

```diff
@@ -0,0 +1,146 @@
+            timeout (int, optional): The time (in seconds) to wait while running this task before a timeout occurs. Defaults to 3600.
+        """
+        self.endpoint = endpoint
+        self.total_load = total_load
+        self.cursor = cursor
+        self.pagesize = pagesize
+        self.date_field = date_field
+        self.start_date = start_date
+        self.end_date = end_date
+
+        super().__init__(
+            name="customer_gauge_to_df",
+            timeout=timeout,
+            *args,
+            **kwargs,
+        )
+
+    def __call__(self):
+        """Download Customer Gauge data to a DF"""
+        super().__call__(self)
+
+    @defaults_from_attrs(
+        "endpoint",
+        "total_load",
+        "cursor",
+        "pagesize",
+        "date_field",
+        "start_date",
+        "end_date",
+    )
+    def run(
+        self,
+        endpoint: Literal["responses", "non-responses"] = None,
+        total_load: bool = True,
+        cursor: int = None,
+        pagesize: int = 1000,
+        date_field: Literal[
+            "date_creation", "date_order", "date_sent", "date_survey_response"
+        ] = None,
+        start_date: datetime = None,
+        end_date: datetime = None,
+        credentials_secret: str = "CUSTOMER-GAUGE",
+        vault_name: str = None,
+    ) -> pd.DataFrame:
+        """
+        Run method. Downloading the selected range of data from Customer Gauge endpoint and return as one pandas DataFrame.
+
+        Args:
+            endpoint (Literal["responses", "non-responses"]): Indicate which endpoint to connect. Defaults to None.
+            total_load (bool, optional): Indicate whether to download the data to the latest. If 'False', only one API call is executed (up to 1000 records). Defaults to True.
+            cursor (int, optional): Cursor value to navigate to the page. Defaults to None.
+            pagesize (int, optional): Number of responses (records) returned per page, max value = 1000. Defaults to 1000.
+            date_field (Literal["date_creation", "date_order", "date_sent", "date_survey_response"], optional): Specifies the date type which filter date range. Defaults to None.
+            start_date (datetime, optional): Defines the period end date in yyyy-mm-dd format. Defaults to None.
+            end_date (datetime, optional): Defines the period start date in yyyy-mm-dd format. Defaults to None.
+            credentials_secret (str, optional): The name of the Azure Key Vault secret containing a dictionary with ['client_id', 'client_secret']. Defaults to "CUSTOMER-GAUGE".
+            vault_name (str, optional): The name of the vault from which to obtain the secret. Defaults to None.
+
+        Returns:
+            pd.DataFrame: Final pandas DataFrame.
+        """
+        try:
+            credentials_str = AzureKeyVaultSecret(
+                credentials_secret, vault_name=vault_name
+            ).run()
+        except ValueError:
+            pass
```

Comment:
Add a message: "Credentials not found." or something similar -  If it's not logged anywhere in the task.

Reply:
I'm not sure about that. If the credentials are not found in KV, then it checks local_config file. It's defined in source class.

Reply:
It is ok if the error message is in the source.
In line 111 you have `credentials = json.loads(credentials_str)` if `credentials_str` is not found (=None) then this operation will return `TypeError: the JSON object must be str, bytes or bytearray, not NoneType`.

Please consider adding line 111 to `try except` block above. Then you can catch the `ValueError and TypeError`